ext/tidy/tidy.c

```diff
@@ -1609,25 +1603,34 @@ PHP_METHOD(tidyNode, isPhp)
 /* {{{ Returns the parent node if available or NULL */
 PHP_METHOD(tidyNode, getParent)
 {
-	TidyNode	parent_node;
-	PHPTidyObj *newobj;
 	TIDY_FETCH_ONLY_OBJECT;
 
-	parent_node = tidyGetParent(obj->node);
-	if(parent_node) {
-		tidy_instantiate(tidy_ce_node, return_value);
-		newobj = Z_TIDY_P(return_value);
-		newobj->node = parent_node;
-		newobj->type = is_node;
-		newobj->ptdoc = obj->ptdoc;
-		newobj->ptdoc->ref_count++;
-		tidy_add_node_default_properties(newobj);
-	} else {
-		ZVAL_NULL(return_value);
```

Comment:
just lost here ... not necessary anymore right ?

Reply:
You never need to set the return value to IS_NULL because the VM & JIT do it prior to the call.
I removed it to reduce the number of lines that are very very similar.